### PR TITLE
refactor: generate fuel types schema

### PR DESCRIPTION
**Pull request by jsolaas**

### CODE REVIEW

File: src/ecalc/libraries/libecalc/common/tests/input/validation/snapshots/test_validation_json_schemas/test_json_schema_changed/schemas.json [hunk 1]
```diff
@@ -489,6 +559,58 @@
...
+                                {
+                                    "type": "integer"
+                                }
+                            ],
+                            "description": "Price or sales value of fuel in NOK per Sm3.\n\nhttps://test.ecalc.equinor.com/docs/docs/modelling/keywords/PRICE",
+                            "title": "PRICE"
+                        }
+                    },
+                    "required": [
+                        "NAME"
```
Comment (jsolaas) [inline src/ecalc/libraries/libecalc/common/tests/input/validation/snapshots/test_validation_json_schemas/test_json_schema_changed/schemas.json:609]: Should require emission, right? ooor?

File: src/ecalc/libraries/libecalc/common/tests/input/validation/snapshots/test_validation_json_schemas/test_json_schema_changed/schemas.json [hunk 2]
```diff
@@ -1266,6 +1391,7 @@
                 }
             },
             "required": [
+                "FUEL_TYPES",
```
Comment (jsolaas) [inline src/ecalc/libraries/libecalc/common/tests/input/validation/snapshots/test_validation_json_schemas/test_json_schema_changed/schemas.json:1394]: Is it required?

Comment (jsolaas) [reply src/ecalc/libraries/libecalc/common/tests/input/validation/snapshots/test_validation_json_schemas/test_json_schema_changed/schemas.json:1395]: yes, fuel for a consumer is required, so therefore this is required as we only support a reference

File: src/ecalc/libraries/libecalc/common/tests/input/validation/snapshots/test_validation_json_schemas/test_json_schema_changed/schemas.json [hunk 3]
```diff
@@ -489,6 +559,59 @@
...
+                                    "type": "integer"
+                                }
+                            ],
+                            "description": "Price or sales value of fuel in NOK per Sm3.\n\nhttps://test.ecalc.equinor.com/docs/docs/modelling/keywords/PRICE",
+                            "title": "PRICE"
+                        }
+                    },
+                    "required": [
+                        "NAME",
+                        "EMISSIONS"
```
Comment (jsolaas) [inline src/ecalc/libraries/libecalc/common/tests/input/validation/snapshots/test_validation_json_schemas/test_json_schema_changed/schemas.json:610]: ecalc raises error if not specified

**Review by frodehk — APPROVED** [review]

**Pull request merged**